spring/src/main/java/io/seata/spring/annotation/datasource/SeataAutoDataSourceProxyAdvice.java

```diff
@@ -53,15 +53,14 @@ public SeataAutoDataSourceProxyAdvice(String dataSourceProxyMode) {
 
     @Override
     public Object invoke(MethodInvocation invocation) throws Throwable {
-        if (!RootContext.requireGlobalLock()
-                && dataSourceProxyMode != RootContext.getBranchType()) {
+        if (!RootContext.requireGlobalLock() && dataSourceProxyMode != RootContext.getBranchType()) {
             return invocation.proceed();
         }
 
         Method method = invocation.getMethod();
         Object[] args = invocation.getArguments();
         Method m = BeanUtils.findDeclaredMethod(dataSourceProxyClazz, method.getName(), method.getParameterTypes());
-        if (m != null) {
+        if (m != null && method.getDeclaringClass().isAssignableFrom(DataSource.class)) {
```

Comment:
DruidDataSource isAssignableFrom DataSource Condition is not established

Reply:
fixed